Update the description of Invalid_State_Err

abdelrazek-alaa opened the pull request:
Fixes #2589
everything as the closed PR #3155
plus, I've made the suggested edit by @wbamberg
>But it might be worth adding an extra paragraph after the introductory paragraph, something like:
 The browser will throw an exception if you call `send()` when the connection is in the `CONNECTING` state. If you call `send()` when the connection is in the `CLOSING` or `CLOSED` states, the browser will silently discard the data.

abdelrazek-alaa committed:
Update the description of Invalid_State_Err

wbamberg committed:
Add an xref to readyState

wbamberg reviewed (approved):
Thank you @abdelrazek-alaa for taking care of this issue!

I added a link to `readyState` and slightly cheekily also removed a note about an ancient version of Firefox.

abdelrazek-alaa commented:
> Thank you @abdelrazek-alaa for taking care of this issue!
> 
> I added a link to `readyState` and slightly cheekily also removed a note about an ancient version of Firefox.

Yay! my first contribution on MDN. that was fun.
@wbamberg you really made everything easy for me. thank you very much :)
